wps_jobs/wps_jobs/weather_model_jobs/ecmwf_prediction_processor.py

```diff
@@ -77,8 +77,9 @@ def _should_interpolate(self, prev_prediction: ModelRunPrediction, prediction: M
             # Timestamps on the same day but surround 20:00 UTC should be interpolated
             return prev_timestamp.hour <= 18 and next_timestamp.hour >= 21
         
-        # Otherwise the next timestamp is on the next day and should always be interpolated for 20:00 UTC
-        return True
+        # datetimes are on different days, interpolate if next is after 20:00 UTC
+        return next_timestamp.hour > 20
```

Comment:
I agree that the assumptions around hour 18 and hour 21 are a little dangerous and make things error prone. I'd suggest a couple changes to this function to make it more generic and applicable to other weather models that predict hourly.
Although not necessary for ECMWF at the moment, we could add:
`
        if next_timestamp.hour == 20:
            return False
`
This will help protect us if the ECMWF eventually provides hourly updates.

I don't think the final comparison is right. We currently arrive at line 83 in a situation where we have hour 18 today and hour 0 tomorrow. This comparison will never result in interpolation. It might be safer to do:
`return prev_timestamp.hour < 20`.